clang/lib/Driver/Driver.cpp

```diff
@@ -4805,8 +4818,12 @@ Action *Driver::BuildOffloadingActions(Compilation &C,
         TCAndArchs.push_back(std::make_pair(TC, Arch));
     }
 
-    for (unsigned I = 0, E = TCAndArchs.size(); I != E; ++I)
-      DeviceActions.push_back(C.MakeAction<InputAction>(*InputArg, InputType));
+    for (unsigned I = 0, E = TCAndArchs.size(); I != E; ++I) {
+      auto *IA = C.MakeAction<InputAction>(*InputArg, InputType);
+      if (CUIDOpts.isEnabled())
+        IA->setId(CUIDOpts.getCUID(IA->getInputArg().getValue(), Args));
+      DeviceActions.push_back(IA);
+    }
```

Comment:
Nit: The for loop with the loop variables unused looks a bit odd here. Can we populate `DeviceActions` in the loop above, on line 4818 where we populate `TCAndArchs` ?

Reply:
Yes. They are just duplicates for the same input file. Will move this step into the above loops. Also we only need to calculate CUID once since it is for the same input file.